client/jetpack-connect/index.js

```diff
@@ -58,7 +58,16 @@ export default function() {
 
 	if ( isLoggedOut ) {
 		page(
-			'/jetpack/connect/authorize/:locale?',
+			'/jetpack/connect/authorize/:localeOrInterval?',
```

Comment:
We can remove the guesswork from this route def and the corresponding middleware by leveraging `page.js`'s pattern matching to explicitly list all possible intervals:

`/jetpack/connect/authorize/:interval(monthly:yearly)`